src/main/java/de/tum/cit/aet/artemis/core/web/AccountResource.java

```diff
@@ -133,46 +139,87 @@ public ResponseEntity<Void> changePassword(@RequestBody PasswordChangeDTO passwo
     }
 
     /**
-     * PUT account/ssh-public-key : sets the ssh public key
+     * GET account/ssh-public-keys : retrieves all SSH keys of a user
      *
-     * @param sshPublicKey the ssh public key to set
+     * @return the ResponseEntity containing all public SSH keys of a user with status 200 (OK)
+     */
+    @GetMapping("account/ssh-public-keys")
+    @EnforceAtLeastStudent
+    public ResponseEntity<List<UserSshPublicKeyDTO>> getSshPublicKeys() {
+        User user = userRepository.getUser();
+        List<UserSshPublicKeyDTO> keys = userSshPublicKeyService.getAllSshKeysForUser(user);
+        return ResponseEntity.ok(keys);
+    }
+
+    /**
+     * GET account/ssh-public-key : gets the ssh public key
      *
-     * @return the ResponseEntity with status 200 (OK), with status 404 (Not Found), or with status 400 (Bad Request)
+     * @param keyId The id of the key that should be fetched
+     *
+     * @return the ResponseEntity containing the requested public SSH key of a user with status 200 (OK), or with status 403 (Access Forbidden) if the key does not exist or is not
+     *         owned by the requesting user
+     */
+    @GetMapping("account/ssh-public-key/{keyId}")
+    @EnforceAtLeastStudent
+    public ResponseEntity<UserSshPublicKeyDTO> getSshPublicKey(@PathVariable Long keyId) {
+        User user = userRepository.getUser();
+        UserSshPublicKey key = userSshPublicKeyService.getSshKeyForUser(user, keyId);
+        return ResponseEntity.ok(UserSshPublicKeyDTO.of(key));
+    }
+
+    /**
+     * GET account/has-ssh-public-key : gets the ssh public key
+     *
+     * @return the ResponseEntity containing true if the User has SSH keys, and false if it does not, with status 200 (OK)
      */
-    @PutMapping("account/ssh-public-key")
+    @GetMapping("account/has-ssh-public-keys")
     @EnforceAtLeastStudent
-    public ResponseEntity<Void> addSshPublicKey(@RequestBody String sshPublicKey) throws GeneralSecurityException, IOException {
+    public ResponseEntity<Boolean> hasUserSSHkeys() {
+        User user = userRepository.getUser();
+        boolean hasKeys = userSshPublicKeyService.hasUserSSHkeys(user.getId());
+        return ResponseEntity.ok(hasKeys);
+    }
```

Comment:
I feel like this endpoint is a bit repetitive when considering that the `GET account/ssh-public-key/{keyId}` also delivers a comparable functionality (200 when existent, 403 when not existent)

Reply:
I removed that request, as for the code button we need all keys anyway, if we want to know if some are expired